### PR TITLE
MBS-8543: Option added to pass on WLAN data via QR code

### PR DESCRIPTION
**Pull request by fdagner**

### CODE REVIEW

File: lang/en/block_qr.php
Comment (sh-csg) [inline lang/en/block_qr.php:1]: String wlan_passkey is missing (you see that if hovering over the help icon in the edit form).

File: edit_form.php [hunk 1]
```diff
@@ -174,6 +175,43 @@ protected function specific_definition($mform) {
...
+        $mform->hideIf('config_wlanssidoptions', 'config_options', 'neq', 'wlan');
+        $mform->setType('config_wlanssidoptions', PARAM_TEXT);
+        $mform->addElement('text', 'config_wlanpasskey', get_string('passkey_label', 'block_qr'), 'size="20"');
+        $mform->addHelpButton('config_wlanpasskey', 'wlan_passkey', 'block_qr');
+        $mform->hideIf('config_wlanpasskey', 'config_options', 'neq', 'wlan');
+        $mform->setType('config_wlanpasskey', PARAM_NOTAGS);
+        $authenticationoptions = [
+            'nopass' => get_string('none', 'block_qr'),
+            'WEP' => get_string('wep', 'block_qr'),
+            'WPA/WPA2' => get_string('wpa', 'block_qr'),
```
Comment (sh-csg) [inline edit_form.php:202]: Maybe encrypted wifi is the more common use case for this option, so this might be the default?

File: lang/en/block_qr.php
Comment (sh-csg) [inline lang/en/block_qr.php:1]: Strings should be in alphabetical order

File: edit_form.php [hunk 2]
```diff
@@ -82,7 +82,8 @@ protected function specific_definition($mform) {
             'internalcontent' => get_string('internalcontent', 'block_qr'),
             'owncontent' => get_string('owncontent', 'block_qr'),
             'event' => get_string('event', 'block_qr'),
-            'geolocation' => get_string('geolocation', 'block_qr')
+            'geolocation' => get_string('geolocation', 'block_qr'),
+            'wlan' => get_string('wlan', 'block_qr')
```
Comment (sh-csg) [inline edit_form.php]: Comma at the end of a multi-line array

**Review by sh-csg — COMMENTED** [review]
Nice feature, works well!

Maybe you might want to change the variable names from "wlan" to "wifi" as it is the more common expression in English.

**Comment by fdagner** [comment]
Thanks for your feedback. New commits added

**Pull request merged**